explainer.md

```diff
@@ -1,624 +1,397 @@
-    }
-  </style>
-</head>
-<body>
-  <div class="animated" sharedid="header-id" id="header">Shared Element</div>
-  <a href="new-document.html">Click Me</a>
-</body>
-</html>
-```
+The aim of this design is to allow for representations of both Page-A and Page-B to exist at the same time, without the usability, accessibility, or memory concerns of having both complete DOM trees alive.
 
-### New Document
+Here's the example that will be used to explain the design:
 
-```html
-<html>
-<head>
-  <style>
-    body {
-      background-color: lightblue;
-    }
-    .animated {
-      position: relative;
-      top: 100px;
-      width: 200px;
-      height: 200px;
-      background-color: green;
-    }
-
-    ::shared-container(header-id) {
-      animation: ::shared-container-header-id 1s ease-in;
-    }
-  </style>
-</head>
-<body>
-  <div class="animated" sharedid="header-id" id="header">Shared Element</div>
-</body>
-</html>
+<img alt="Page-A and Page-B" src="media/pages.png?raw=true">
+
+The transition between Page-A and Page-B can be a full cross-document navigation between two same-origin pages, or it can be an SPA navigation. The main mechanism and concepts are the same between the two.
+
+Cross-origin transitions are something we want to tackle, but may have significant differences and restrictions for security reasons. Cross-origin transitions are not covered in this document.
+
+## Part 1: The offering
+
+Before Page-A goes away, it offers up parts of itself to be used in the transition. Generally, this will mean one part per 'thing' that acts independently during the transition. For the example transition, the parts are:
+
+- The header
+- The share button
+- The rest (referred to as the page root)
+
+https://user-images.githubusercontent.com/93594/141104275-6d1fb67a-2f73-41e4-9cef-14676798223b.mp4
+
+Aside from the root, an element can only be offered as a transition part if it has paint containment, which clips child content to the content box, although this can be expanded with `overflow-clip-margin`. The is a compromise made to greatly simplify implementation.
+
+Additionally, the element must be a single rect. As in, it doesn't break across lines or columns.
+
+- Open question: If the developer tries to break the above rules, what should happen? Is paint containment applied by the browser? Is that element dropped from the transition? Is the whole transition abandoned?
+
+When a developer offers part of the page for transitions, there are two modes they can choose from:
+
+### As a single texture
+
+The entire painting of the element is captured, including things which appear outside of its bounding box such as shadows and blurs, as a single texture at device-pixel resolution.
+
+https://user-images.githubusercontent.com/93594/141118353-d62d19a1-0964-4fa0-880f-bdde656ce899.mp4
+
+- Open question: Which effects from a parent element are baked into this texture? E.g. things like `filter`, `opacity`, `transform`.
```

Comment:
We capture it at 0deg and reapply the transform to the container. The transform we're using currently is a squashed matrix which includes a translation for the element's layout position + transforms from the ancestor chain. Ian actually brought up a good point that we should use a list of transform operations instead of a single matrix so 3-d transforms work correctly (https://chromium-review.googlesource.com/c/chromium/src/+/3271751/1..4/third_party/blink/renderer/core/document_transition/document_transition_container_element.cc#b154).

But in either case, it will be a transform on the container and won't apply to the image.